include/odp/api/spec/crypto_types.h

```diff
@@ -753,9 +753,11 @@ typedef struct odp_crypto_packet_op_param_t {
 	 *  creation.
 	 *
 	 *  Ignored by the null cipher with operation types other than
-	 *  ODP_CRYPTO_OP_TYPE_OOP. Must be set to zero range (zero offset
-	 *  and zero length) with the null cipher used with the out-of-place
-	 *  operation type.
+	 *  ODP_CRYPTO_OP_TYPE_OOP. In the OOP operation type the cipher
+	 *  range is copied in the output packet even with the null cipher.
+	 *  If an ODP implementation does not support non-zero-length
+	 *  ranges with the null cipher and the OOP operation type, the
+	 *  crypto operation may fail.
```

Comment:
Would it be more precise to say "... operation may report failure.". I.e. to be explicit that operation return code reports a failure rather than there would be undefined behavior.

Reply:
I did not change this yet since we probably need to clarify it more than that. Specifically, if we allow failure, do we allow a failure of odp_crypto_op() / odp_crypto_op_enq() so that the input packet is not even consumed, or a failure reported through the crypto op result, or both. Let's discuss it today.

Reply:
Reworded in v5.